### PR TITLE
Mock the rest of AWS in tests

### PR DESCRIPTION
**Pull request by jmsanders**
With this change, all dagster-aws tests (except for those hiding behind environment variable flags that require live resources) can run locally using mocked resources. This means https://github.com/dagster-io/dagster/issues/2302 can be resolved and hopefully it will make it easier for contributors to extend dagster-aws.

I didn't touch any implementation code. I rewrote a few tests to be a little simpler but otherwise, this is a pure refactor of test code to use the mock fixtures instead of the combination of real resources, magic mocks, repeated uses of the `@mock_s3` decorator that they were already using. I do think there remains a lot of opportunity to improve the tests - as well as some of the implementations - of a lot of dagster-aws to make them more consistent, simple, and readable but those are all tasks for another day 😉

### CODE REVIEW

File: python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+@pytest.fixture
+def mock_ecs_client():
+    with mock_ecs():
+        yield boto3.client("ecs", region_name="us-east-2")
+
+
+@pytest.fixture
+def mock_ecs_cluster(mock_ecs_client):  # pylint: disable=redefined-outer-name
+    name = "test-cluster"
+    mock_ecs_client.create_cluster(clusterName="test-cluster")
```
Comment (catherinewu) [inline python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/conftest.py:15]: nit: `clusterName="test-cluster"` -> `clusterName=name`

File: python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/test_ecs_client.py [hunk 2]
```diff
@@ -4,107 +4,62 @@
...
 
-@mock_ecs
-@mock_logs
-def test_set_cluster():
-    log_client = boto3.client("logs", region_name="us-east-2")
-    ecs_client = boto3.client("ecs", region_name="us-east-2")
-    ecs_client.create_cluster(clusterName="test_clust")
-    testclient = client.ECSClient(
+@pytest.fixture
+def test_client(mock_ecs_client, mock_ecs_cluster):  # pylint: disable=unused-argument
```
Comment (catherinewu) [inline python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/test_ecs_client.py:11]: move this up to ecs_tests/conftest.py? is `mock_ecs_cluster` here for the side effect of creating the cluster?

File: python_modules/libraries/dagster-aws/dagster_aws_tests/emr_tests/test_pyspark.py [hunk 3]
```diff
@@ -124,7 +126,7 @@ def test_pyspark_emr(mock_is_emr_step_complete, mock_read_events):
             "SlaveInstanceType": "c3.xlarge",
         },
         JobFlowRole="EMR_EC2_DefaultRole",
-        LogUri="s3://mybucket/log",
+        LogUri="s3://my/log",
```
Comment (catherinewu) [inline python_modules/libraries/dagster-aws/dagster_aws_tests/emr_tests/test_pyspark.py:129]: nit: "s3://{bucket}/log".format(bucket=mock_s3_bucket.name)?

**Review by catherinewu — COMMENTED** [review]
This is really awesome work 🙌, thanks for substantially improving the dagster-aws test suite!

File: python_modules/libraries/dagster-aws/dagster_aws_tests/emr_tests/test_pyspark.py (hunk 3 above)
Comment (jmsanders) [reply python_modules/libraries/dagster-aws/dagster_aws_tests/emr_tests/test_pyspark.py:129]: Oh yeah, looks like there are actually no assertions that depend on this being set correctly so my errant find/replace didn't cause issues 😐 .

File: python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/test_ecs_client.py (hunk 2 above)
Comment (jmsanders) [reply python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/test_ecs_client.py:11]: Now that you mention it, I might collapse conftest back into the test file. The ECS resources are pretty under-tested so originally I was planning to add some additional test coverage. But if that happens, it'll be in a separate PR anyway.

And yup, `mock_ecs_cluster` is included for the side effect of setup/teardown of the cluster for any test that uses the `test_client`.

**Pull request merged**